src/CMakeLists.txt

```diff
@@ -114,10 +116,28 @@ if(TREXIO_DEVEL)
 	  ${TREXIO_PRIVATE_HEADERS}
 	  ${TREXIO_MOD_FILE}
       COMMAND ./build_trexio.sh
-      DEPENDS ${ORG_FILES}
+      DEPENDS ${ORG_FILES} ${CMAKE_SOURCE_DIR}/include/config.h
       WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/tools
       COMMENT "Generating TREXIO source code from org-mode files."
       VERBATIM)
+
+  set_source_files_properties(${TREXIO_SOURCES}
+                              ${TREXIO_PUBLIC_HEADERS}
+                              ${TREXIO_PRIVATE_HEADERS}
+                              ${TREXIO_MOD_FILE}
+                              PROPERTIES GENERATED TRUE)
+
+  add_custom_command(
```

Comment:
CMake has a built-in `configure_file` function for `config.h.in`-like files. It takes a template file as an input with placeholders for variables to be replaced at build time.

Essentially, lines 130-140 in the current `CMakeLists.txt` are equivalent to the following line in the root `CMakeLists.txt` (at line 46 before `endif`)
- `configure_file(${CMAKE_SOURCE_DIR}/include/cmake_config.h.in ${CMAKE_SOURCE_DIR}/include/config.h)`

with the following `cmake_config.h.in` template file in the `include/` directory of the root: 
```
#define PACKAGE_VERSION "@PROJECT_VERSION@"
#define VERSION "@PROJECT_VERSION@"
#define VERSION_MAJOR @PROJECT_VERSION_MAJOR@
#define VERSION_MINOR @PROJECT_VERSION_MINOR@
#define VERSION_PATCH @PROJECT_VERSION_PATCH@
```

What do you think @plopezrios ?

Reply:
That is of course better, thanks!

Reply:
You meant line 36 not 46 I assume?

Reply:
Yes, sorry, I would even say line 35, or basically here:
```
  else()
    message(FATAL_ERROR "EMACS not found. It is required to produce TREXIO source code from org-mode files.")
  endif()
  configure_file(${CMAKE_SOURCE_DIR}/include/cmake_config.h.in ${CMAKE_SOURCE_DIR}/include/config.h)
endif()
```